Data Visualisation.py

```diff
@@ -7,21 +7,44 @@
 import seaborn as sns
 from tqdm import tqdm
 
-df = pd.read_csv("C:\\Users\\lukvi\\Python Fun\\Data Visualisation\\NBA Data\\triple_doubles.csv")
-df["season"] = df["season"].astype("string").transform(lambda x: x[-2:])
-df_stats = df[["MIN","PTS","REB","AST","STL","BLK"]]
-# print(df.head())
-
-# TD leaders
-df_counts = df.groupby("player",as_index=False).apply(lambda x: x)["player"].value_counts()
-#df_leaders = df.loc[df_counts[0,:5]]
-print(df_counts.head())
-
-df_print = df.groupby("season",as_index=True,sort=False).apply(lambda x: x.count())
-df_stats_print = df.groupby("player",as_index=True,sort=False).apply(lambda x: x.describe())[["MIN","PTS","REB","AST","STL","BLK"]]
-# df_stats_leaders_summary = df_leaders.groupby("player",as_index=True,sort=False).apply(lambda x: x.describe())[["MIN","PTS","REB","AST","STL","BLK"]]
-print(df_stats_print)
-
-# sns.pairplot(df_stats)
+# Loading the data
+td = pd.read_csv("C:\\Users\\lukvi\\Python Fun\\NBA-Data\\triple_doubles.csv") 
+# Simplify Player names and seasoon name. Beware that player names could lead to unclerity, and seasons like this when ordered get mismatched
+td["season"] = td["season"].astype("string").transform(lambda x: x[-2:])
+td["player"] = td["player"].astype("string").transform(lambda x: x.partition(" ")[2])
+
+# The leaders' TDs
+num_leaders = 8
+td_counts = td["player"].value_counts()
+td_leaders = td.groupby("player",as_index=True,sort=False).apply(lambda x: x)[["MIN","PTS","REB","AST","STL","BLK"]]
+td_leaders = td_leaders.loc[td_counts.index[0:num_leaders].to_list()]
+td_leaders = td_leaders.reset_index().drop("level_1",axis=1)
+
+# TD stats for all TDs
+td_stats = td.groupby("player",as_index=True,sort=False).apply(lambda x: x.describe())[["MIN","PTS","REB","AST","STL","BLK"]]
+td_means = td_stats.reset_index()[td_stats.reset_index()["level_1"] == "mean"]
+
+# TD leaders' means
+td_top_5 = td_stats.loc[td_counts.index[0:num_leaders].to_list()]
+td_top_5_means = td_top_5.reset_index()[td_top_5.reset_index()["level_1"]=="mean"]
+
+
+# Boxplots comparing the means of all players TD stats
```

Comment:
possibly add the different plots (including the ones commented out) as functions so these don't have to be commented out?